content/docs/reference/design-docs/conflict-detection/index.en.md

```diff
@@ -0,0 +1,139 @@
+ - it cannot rely on data we do not have
+ - it has to enable later support of mobile block systems
+ - it has to rely on existing signaling and interlocking simulation
+ - it has to enable detecting conflicts regardless of the signaling system in use
+ - support transitions between signaling systems
+ - support conflicts between different signaling systems
+
+## Conflict theory
+
+Conflict detection relies on an abstract model:
+ - objects where conflicts happen are resources
+ - resources are used by actors
+ - resource uses can be incompatible
+ - overlapping incompatible ressource uses generate a conflict
+
+### Actors
+
+Actors are objects which cause resources to be used:
+ - **train paths** (or someone / something on the behalf of the train)
+ - **planned infrastructure work**
+
+### Resources
+
+Resources have the following properties:
+ - can be used by actors
+ - have multiple states
+ - can only be in a single state at any given time
+ - may take time changing state
+
+The current model only considers the following two resources:
+ - **zones**, which have one state per way to traverse it
+ - **switches**, which have one state per position
+
+{{% pageinfo color="info" %}}
+Station platforms could also be resources, which would enable modeling platform use conflicts: if two tracks share a tiny platform, it may not be allowed to use both sides of the platform at once.
+{{% /pageinfo %}}
+
+### Requirements and conflicts
+
+A **requirement** describes what resource is needed by an actor, and when.
+
+There are a few types of requirements:
+ - **shareable**: the resource is required to be in a given configuration, for a given time lapse. Other shareable requirements with the same configuration can stack up.
+ - **exclusive**: the resource is required to be in a given configuration, for a given actor, for a given time lapse. This requirement cannot be combined with others.
+
+**Conflicts** are both real-life disturbances and the incompatible resource requirements which cause them. A disturbance is something that would cause a train to be unable to run as planned.
+ - route setup delays due to unavailable resources
+ - spacing conflicts due to a train catching up with another
+
+These real-life issues can be linked to conflicts, which arise from incompatible resource requirements:
+ - an exclusive requirement overlaps another requirement
+ - a shareable requirements overlaps an incompatible requirement
+ - a resource does not have time to change state between non-overlapping, yet close, requirements
+
+For example, there are **no** conflicts when:
+ - a resource has non-overlapping exclusive requirements
+ - a resource is shared by requirements which require the same configuration
+
+## Generating requirements
+
+**For conflict detection to work, resource requirements have to strictly match what's required
```

Comment:
```suggestion
**In order for conflict detection to work, resource requirements have to strictly match what's required
```
A quick change that makes the sentence a bit cleaner in my opinion (cause it's written and not spoken)

Reply:
[I had never heard this sequence of word before](https://books.google.com/ngrams/graph?content=In+order+for%2CIn+order+to%2Cso+that%2C+for+this&year_start=1800&year_end=2019&corpus=en-2019&smoothing=3), and would rather leave it as it if you do not mind